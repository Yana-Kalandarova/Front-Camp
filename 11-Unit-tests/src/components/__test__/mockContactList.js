import mockContact from './mockContact';

const mockContactList = [
  mockContact,
  mockContact,
];

export default mockContactList;
