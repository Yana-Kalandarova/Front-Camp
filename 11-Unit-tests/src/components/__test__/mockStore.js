import React from 'react';
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore({ contactList: [], searchValue: '' })

export default store;
