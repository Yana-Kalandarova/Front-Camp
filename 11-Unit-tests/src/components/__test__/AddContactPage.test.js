import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import AddContactPage from '../AddContactPage';
import mockStore from './mockStore';
import mockContactList from './mockContactList';

describe('AddContactPage', () => {
  it('renders correctly', () => {
    expect(
      mount(
        <Provider store={mockStore}>
          <BrowserRouter>
            <AddContactPage onUpdateContact={jest.fn()} contactList={mockContactList} />
          </BrowserRouter>
        </Provider>
      )
    ).toMatchSnapshot()
  })
})
