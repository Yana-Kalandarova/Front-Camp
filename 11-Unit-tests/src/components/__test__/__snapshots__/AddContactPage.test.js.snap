// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddContactPage renders correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(AddContactPage)
        contactList={
          Array [
            Object {
              "_id": "1a2b3c",
              "name": "Test Name",
              "phoneNumber": "12345",
            },
            Object {
              "_id": "1a2b3c",
              "name": "Test Name",
              "phoneNumber": "12345",
            },
          ]
        }
        onUpdateContact={[MockFunction]}
      >
        <AddContactPage
          contactList={Array []}
          onAddContact={[Function]}
          onUpdateContact={[MockFunction]}
        >
          <div
            className="view_wrap"
          >
            <h1
              className="view_title"
            >
              Add Contact
            </h1>
            <withRouter(ContactForm)
              contactList={Array []}
              onUpdateContact={[Function]}
            >
              <ContactForm
                contactList={Array []}
                history={
                  Object {
                    "action": "POP",
                    "block": [Function],
                    "createHref": [Function],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "length": 1,
                    "listen": [Function],
                    "location": Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    },
                    "push": [Function],
                    "replace": [Function],
                  }
                }
                location={
                  Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  }
                }
                match={
                  Object {
                    "isExact": true,
                    "params": Object {},
                    "path": "/",
                    "url": "/",
                  }
                }
                onUpdateContact={[Function]}
              >
                <form
                  className="contact_form"
                  onSubmit={[Function]}
                >
                  <div
                    className="contact_form-field-wrap"
                  >
                    <label
                      htmlFor="name"
                    >
                      Name:
                    </label>
                    <input
                      className="contact_form-field"
                      id="name"
                      name="name"
                      onChange={[Function]}
                      placeholder="Name"
                      type="text"
                      value=""
                    />
                    <ErrorMessage
                      message=""
                    />
                  </div>
                  <div
                    className="contact_form-field-wrap"
                  >
                    <label
                      htmlFor="phoneNumber"
                    >
                      Phone Number:
                    </label>
                    <input
                      className="contact_form-field"
                      id="phoneNumber"
                      name="phoneNumber"
                      onChange={[Function]}
                      placeholder="Phone"
                      type="tel"
                      value=""
                    />
                    <ErrorMessage
                      message=""
                    />
                  </div>
                  <div
                    className="contact_btn-wrap"
                  >
                    <Link
                      className="contact_btn-cancel"
                      to="/"
                    >
                      <LinkAnchor
                        className="contact_btn-cancel"
                        href="/"
                        navigate={[Function]}
                      >
                        <a
                          className="contact_btn-cancel"
                          href="/"
                          onClick={[Function]}
                        >
                          Cancel
                        </a>
                      </LinkAnchor>
                    </Link>
                    <button
                      className="contact_btn-save"
                      disabled={true}
                      type="submit"
                    >
                      Save
                    </button>
                  </div>
                </form>
              </ContactForm>
            </withRouter(ContactForm)>
          </div>
        </AddContactPage>
      </Connect(AddContactPage)>
    </Router>
  </BrowserRouter>
</Provider>
`;
